src/extensions/extensionsApi.ts

```diff
@@ -388,14 +389,19 @@ export async function configureInstance(args: {
       },
     },
   };
-  if (args.allowedEventTypes !== undefined) {
+  if (!args.canEmitEvents) {
+    reqBody.data.config.allowedEventTypes = undefined;
+    reqBody.data.config.eventarcChannel = undefined;
```

Comment:
Isn't it unnecessary to explicitly set these to undefined here? IIUC correctly, just including the fields in updateMask should be sufficient